src/Agent.Worker/WorkerCommandManager.cs

```diff
@@ -145,6 +153,42 @@ public interface IWorkerCommand
         void Execute(IExecutionContext context, Command command);
     }
 
+    public interface IWorkerCommandRestrictionPolicy
+    {
+        bool isCommandAllowed(IWorkerCommand command);
+    }
+
+    public class UnrestricedWorkerCommandRestrictionPolicy: IWorkerCommandRestrictionPolicy
+    {
+        public bool isCommandAllowed(IWorkerCommand command)
+        {
+            return true;
+        }
+    }
+
+    [ AttributeUsage(AttributeTargets.Class, Inherited = false, AllowMultiple = false)]
+    public sealed class CommandRestrictionAttribute : Attribute
+    {
+        public bool AllowedInRestrictedMode { get; set; }
+    }
+
+    public class AttributeBasedWorkerCommandRestrictionPolicy: IWorkerCommandRestrictionPolicy
```

Comment:
This policy is not used anywhere yet. When we fully implement this feature, when we get to a step that wants to restrict commands, it should set the command policy to an instance of this object.